<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Parses and analyses output of tandem repeat detection software.">
<title>Analyse tandem repeats • changchang</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Analyse tandem repeats">
<meta property="og:description" content="Parses and analyses output of tandem repeat detection software.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">changchang</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/changchang.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="changchang-manual">ChangChang manual<a class="anchor" aria-label="anchor" href="#changchang-manual"></a>
</h1></div>
<!-- README.md is generated from README.Rmd. Please edit that file -->
</div>
<div class="section level1">
<h1 id="changchang">changchang<a class="anchor" aria-label="anchor" href="#changchang"></a>
</h1>
<!-- badges: start -->

<p>The goal of <code>changchang</code> is to …</p>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/" class="external-link">CRAN</a>. Then install <code>changchang</code> from <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> using the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"changchang"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h2>
<div class="section level3">
<h3 id="tantan">
<code>tantan</code><a class="anchor" aria-label="anchor" href="#tantan"></a>
</h3>
<p>To load output of <code>tantan</code> the simplest at the moment is to convert its output to BED format and use the <code><a href="reference/loadTantan.html">loadTantan()</a></code> function. Here is an exmaple on how to produce the BED file.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>tantan <span class="sc">-</span>w <span class="dv">10</span> <span class="sc">-</span>f4 example.fasta <span class="sc">|</span> tee example.tantan.txt <span class="sc">|</span> awk <span class="st">'OFS="</span><span class="sc">\t</span><span class="st">" {print $1, $2, $3, $6, $5, "."}'</span>  <span class="sc">&gt;</span> example.tantan.bed</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="other-tandem-repeat-finder-softwares">Other tandem repeat finder softwares<a class="anchor" aria-label="anchor" href="#other-tandem-repeat-finder-softwares"></a>
</h3>
<p>Not supported yet. Patches welcome!</p>
</div>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>This is a basic example which shows you how to solve a common problem:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"changchang"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/loadTantan.html">loadTantan</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">'extdata/example.bed.xz'</span>, package<span class="op">=</span><span class="st">"changchang"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 1000 ranges and 3 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames          ranges strand |     score</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;       &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;      [1] 1086a8205b           1-613      + | 102.66700</span></span>
<span><span class="co">#&gt;      [2] 1086a8205b         645-767      + |   3.42857</span></span>
<span><span class="co">#&gt;      [3] 1086a8205b        766-6420      + | 106.67900</span></span>
<span><span class="co">#&gt;      [4] 1086a8205b       7381-7473      + |   2.26829</span></span>
<span><span class="co">#&gt;      [5] 1086a8205b       8178-8206      - |   2.23077</span></span>
<span><span class="co">#&gt;      ...        ...             ...    ... .       ...</span></span>
<span><span class="co">#&gt;    [996] 1086a8205b 1706037-1706047      - |   5.50000</span></span>
<span><span class="co">#&gt;    [997] 1086a8205b 1706066-1706079      - |   2.80000</span></span>
<span><span class="co">#&gt;    [998] 1086a8205b 1706963-1706986      + |   3.25000</span></span>
<span><span class="co">#&gt;    [999] 1086a8205b 1707022-1707042      + |   4.20000</span></span>
<span><span class="co">#&gt;   [1000] 1086a8205b 1707081-1707106      + |   2.16667</span></span>
<span><span class="co">#&gt;                                                     normTandem     nchar</span></span>
<span><span class="co">#&gt;                                                       &lt;factor&gt; &lt;integer&gt;</span></span>
<span><span class="co">#&gt;      [1] AACCCT                                                        6</span></span>
<span><span class="co">#&gt;      [2] AACCCTAACCCTAACCCTAACCCTAACCCTAACCT                          35</span></span>
<span><span class="co">#&gt;      [3] AAACCTAGCGACAACACAGGGAGAACACTCTCACCTTAACAAGAACCCTTATT        53</span></span>
<span><span class="co">#&gt;      [4] AAAATTTTGCGCGGTTGCGGGCGGGCGCGGGGGCCGATTCC                    41</span></span>
<span><span class="co">#&gt;      [5] ACGGGCGGGGGCC                                                13</span></span>
<span><span class="co">#&gt;      ...                                                   ...       ...</span></span>
<span><span class="co">#&gt;    [996]                                          CG                   2</span></span>
<span><span class="co">#&gt;    [997]                                          AAAAC                5</span></span>
<span><span class="co">#&gt;    [998]                                          ACTAGCC              7</span></span>
<span><span class="co">#&gt;    [999]                                          AGCCT                5</span></span>
<span><span class="co">#&gt;   [1000]                                          AAAAAACTAGCC        12</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/telomeres.html">telomeres</a></span><span class="op">(</span><span class="va">exampleTantan</span>, narrow<span class="op">=</span><span class="st">"ends"</span><span class="op">)</span></span>
<span><span class="co">#&gt; GRanges object with 1 range and 5 metadata columns:</span></span>
<span><span class="co">#&gt;         seqnames    ranges strand |     score normTandem     nchar    rstart</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;   &lt;factor&gt; &lt;integer&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1] 1086a8205b     1-613      + |   102.667     AACCCT         6   2232853</span></span>
<span><span class="co">#&gt;            rend</span></span>
<span><span class="co">#&gt;       &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;   [1]   2233465</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 1 sequence from example genome</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Below is the citation output from using <code>citation('changchang')</code> in R. Please run this yourself to check for any updates on how to cite <strong>changchang</strong>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">'changchang'</span><span class="op">)</span>, bibtex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; To cite package 'changchang' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Plessy C (2024). _ChangChang_. doi:10.18129/B9.bioc.changchang</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.18129/B9.bioc.changchang&gt;,</span></span>
<span><span class="co">#&gt;   https://github.com//changchang - R package version 0.99.0,</span></span>
<span><span class="co">#&gt;   &lt;http://www.bioconductor.org/packages/changchang&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {ChangChang},</span></span>
<span><span class="co">#&gt;     author = {Charles Plessy},</span></span>
<span><span class="co">#&gt;     year = {2024},</span></span>
<span><span class="co">#&gt;     url = {http://www.bioconductor.org/packages/changchang},</span></span>
<span><span class="co">#&gt;     note = {https://github.com//changchang - R package version 0.99.0},</span></span>
<span><span class="co">#&gt;     doi = {10.18129/B9.bioc.changchang},</span></span>
<span><span class="co">#&gt;   }</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Plessy C (2024). "ChangChang." _bioRxiv_. doi:10.1101/TODO</span></span>
<span><span class="co">#&gt;   &lt;https://doi.org/10.1101/TODO&gt;,</span></span>
<span><span class="co">#&gt;   &lt;https://www.biorxiv.org/content/10.1101/TODO&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Article{,</span></span>
<span><span class="co">#&gt;     title = {ChangChang},</span></span>
<span><span class="co">#&gt;     author = {Charles Plessy},</span></span>
<span><span class="co">#&gt;     year = {2024},</span></span>
<span><span class="co">#&gt;     journal = {bioRxiv},</span></span>
<span><span class="co">#&gt;     doi = {10.1101/TODO},</span></span>
<span><span class="co">#&gt;     url = {https://www.biorxiv.org/content/10.1101/TODO},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
<p>Please note that the <code>changchang</code> was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the <code>changchang</code> project is released with a <a href="http://bioconductor.org/about/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
<div class="section level2">
<h2 id="development-tools">Development tools<a class="anchor" aria-label="anchor" href="#development-tools"></a>
</h2>
<ul>
<li>Continuous code testing is possible thanks to <a href="https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/" class="external-link">GitHub actions</a> through <em><a href="https://CRAN.R-project.org/package=usethis" class="external-link">usethis</a></em>, <em><a href="https://CRAN.R-project.org/package=remotes" class="external-link">remotes</a></em>, and <em><a href="https://CRAN.R-project.org/package=rcmdcheck" class="external-link">rcmdcheck</a></em> customized to use <a href="https://www.bioconductor.org/help/docker/" class="external-link">Bioconductor’s docker containers</a> and <em><a href="https://bioconductor.org/packages/3.19/BiocCheck" class="external-link">BiocCheck</a></em>.</li>
<li>Code coverage assessment is possible thanks to <a href="https://codecov.io/gh" class="external-link">codecov</a> and <em><a href="https://CRAN.R-project.org/package=covr" class="external-link">covr</a></em>.</li>
<li>The code is styled automatically thanks to <em><a href="https://CRAN.R-project.org/package=styler" class="external-link">styler</a></em>.</li>
<li>The documentation is formatted thanks to <em><a href="https://CRAN.R-project.org/package=devtools" class="external-link">devtools</a></em> and <em><a href="https://CRAN.R-project.org/package=roxygen2" class="external-link">roxygen2</a></em>.</li>
</ul>
<p>For more details, check the <code>dev</code> directory.</p>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.19/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://support.bioconductor.org/tag/changchang" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/AGPL-3" class="external-link">AGPL-3</a></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
<li><a href="SUPPORT.html">Getting help</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing changchang</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Charles Plessy <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-7410-6295" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental" class="external-link"><img src="https://img.shields.io/badge/lifecycle-experimental-orange.svg" alt="Lifecycle: experimental"></a></li>
<li><a href="https://bioconductor.org/checkResults/release/bioc-LATEST/changchang" class="external-link"><img src="http://www.bioconductor.org/shields/build/release/bioc/changchang.svg" alt="Bioc release status"></a></li>
<li><a href="https://bioconductor.org/checkResults/devel/bioc-LATEST/changchang" class="external-link"><img src="http://www.bioconductor.org/shields/build/devel/bioc/changchang.svg" alt="Bioc devel status"></a></li>
<li><a href="http://bioconductor.org/packages/stats/bioc/changchang/" class="external-link"><img src="https://bioconductor.org/shields/downloads/release/changchang.svg" alt="Bioc downloads rank"></a></li>
<li><a href="https://support.bioconductor.org/tag/changchang" class="external-link"><img src="https://bioconductor.org/shields/posts/changchang.svg" alt="Bioc support"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/changchang.html#since" class="external-link"><img src="https://bioconductor.org/shields/years-in-bioc/changchang.svg" alt="Bioc history"></a></li>
<li><a href="http://bioconductor.org/checkResults/devel/bioc-LATEST/changchang/" class="external-link"><img src="https://bioconductor.org/shields/lastcommit/devel/bioc/changchang.svg" alt="Bioc last commit"></a></li>
<li><a href="https://bioconductor.org/packages/release/bioc/html/changchang.html#since" class="external-link"><img src="https://bioconductor.org/shields/dependencies/release/changchang.svg" alt="Bioc dependencies"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Charles Plessy.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
